src/Chainweb/Pact/Templates.hs

```diff
@@ -0,0 +1,107 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TupleSections #-}
+{-# LANGUAGE TypeApplications #-}
+
+-- |
+-- Module      :  Chainweb.Pact.Templates
+-- Copyright   :  Copyright © 2010 Kadena LLC.
+-- License     :  (see the file LICENSE)
+-- Maintainer  :  Stuart Popejoy
+-- Stability   :  experimental
+--
+-- Prebuilt Term templates for automated operations (coinbase, gas buy)
+--
+module Chainweb.Pact.Templates
+  ( mkBuyGasTerm
+  , mkCoinbaseTerm
+  ) where
+
+
+import Control.Lens
+import Data.Aeson hiding ((.=))
+import qualified Data.Aeson as A
+import Data.Default (def)
+import Data.Text (Text)
+
+import Pact.Parse
+import Pact.Types.Command
+import Pact.Types.RPC
+import Pact.Types.Runtime
+
+import Chainweb.Miner.Pact
+import Chainweb.Pact.Types
+
+app :: Name -> [Term Name] -> Term Name
+app f as = TApp (App (TVar f def) as def) def
```

Comment:
we can inline `app` `qn`, `bn` and `strLit`

Reply:
INLINABLE?

Reply:
`INLINE` would be fine for these